smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -147,9 +147,19 @@ contract MixinKeys is
   )
     public
     view
-    returns (bool)
-  {
-    return keyByOwner[_keyOwner].expirationTimestamp > block.timestamp;
+    returns (bool isValid)
+  { 
+    isValid = keyByOwner[_keyOwner].expirationTimestamp > block.timestamp;
+
+    // use hook if it exists
+    if(address(onValidKeyHook) != address(0)) {
+      isValid = onValidKeyHook.hasValidKey(
+        address(this),
+        _keyOwner,
+        keyByOwner[_keyOwner].expirationTimestamp,
+        isValid
+      );
+    }    
   }
```

Comment:
Are we missing a return statement?

Reply:
nope. It turns out that Solidity will declare and return the variable named in the `returns` declaration, so the `returns (bool isValid)` is enough for to return it. I have seen it used elsewhere in the Unlock contracts already